### ISSUE TITLE
Existing MTG Card Art Pulled for Custom Card with Similar Name

### ISSUE THREAD

**Issue by slumberspirit** [Cockatrice/Cockatrice#4210]
I created a custom set file and added imgur links for each card, and Cockatrice successfully displays them for all 36 cards other than one named "Godhead." For that one, the art displayed is for the existing Magic card "Godhead of Awe." I tried deleting the art for "Godhead of Awe" from my computer but it just pulls it again when I select the custom card "Godhead." My imgur link for the "Godhead" art I want is working properly. I have my custom set pulled to the top of the art preference list in the Manage Sets menu.

The "URL" spot below is where I'd been entering the imgur URLs within the custom set file for each card:
<set rarity="mythic rare" uuid="" num="" muid="" picurl="URL">CUBE</set>

**Comment by ebbit1q** [comment]
cockatrice only tries to find a missing image after it failed to download your custom one, there is probably something wrong with either the set or the image, could you include your log files when the issue occurs (like the issue template instructs you to)

here is an excerpt from my reproduction of the issue (without set):
```
PictureLoader: [card:  "godhead"  set:  "" ]: Requested property ( "side" ) for Url template ( "https://api.scryfall.com/cards/!set:uuid!?format=image&face=!prop:side!" ) is not available
PictureLoader: [card:  "godhead"  set:  "" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "godhead"  set:  "" ]: Requested set property ( "muid" ) for Url template ( "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "godhead"  set:  "" ]: Trying to load picture
PictureLoader: [card:  "godhead"  set:  "" ]: Picture not found on disk, trying to download
PictureLoader: [card:  "godhead"  set:  "" ]: Trying to download picture from url: "http://gatherer.wizards.com/Handlers/Image.ashx?name=godhead&type=card"
PictureLoader: [card:  "godhead"  set:  "" ]: following redirect: "https://gatherer.wizards.com/Handlers/Image.ashx?name=godhead&type=card"
PictureLoader: [card:  "godhead"  set:  "" ]: Image successfully downloaded from  "https://gatherer.wizards.com/Handlers/Image.ashx?name=godhead&type=card"
```

**Comment by ebbit1q** [comment]
here is what the log will say if it can't find the picurl:
```
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Requested property ( "side" ) for Url template ( "https://api.scryfall.com/cards/!set:uuid!?format=image&face=!prop:side!" ) is not available
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Requested set property ( "muid" ) for Url template ( "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Trying to load picture
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Picture not found on disk, trying to download
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Trying to download picture from url: "https://github.com/Cockatrice/Cockatrice/issues/4210"
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Possible picture at  "https://github.com/Cockatrice/Cockatrice/issues/4210"  could not be loaded
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Trying to load picture
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Picture not found on disk, trying to download
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Trying to download picture from url: "http://gatherer.wizards.com/Handlers/Image.ashx?name=godhead&type=card"
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: following redirect: "https://gatherer.wizards.com/Handlers/Image.ashx?name=godhead&type=card"
PictureLoader: [card:  "godhead"  set:  "AAAAA" ]: Image successfully downloaded from  "https://gatherer.wizards.com/Handlers/Image.ashx?name=godhead&type=card"
```

**Comment by ebbit1q** [comment]
do note that after fixing the picurl you might have to remove this already downloaded picture, we should consider turning image caching off for images that are downloaded after failing to download from picurl in the application

**Comment by slumberspirit** [comment]
Thank you for your response! I got it to work. Previously I'd tried deleting the Godhead of Awe art from my SHM (Shadowmoor) folder in downloadedpics. But I just found that Cockatrice had downloaded another image of the card Godhead of Awe to my custom set folder "CUBE" in downloadedpics and just called the file Godhead. I deleted that and then when I restarted Cockatrice and searched for Godhead, it pulled the right image from the imgur link and downloaded it to the "CUBE" folder in downloadedpics. It's just weird, because I didn't change the set file or image URL, so I don't know why it downloaded a Godhead of Awe image and called it Godhead the first time.

Here's the new, working properly log if you're curious.


PictureLoader: [card:  "Godhead"  set:  "CUBE" ]: Requested set property ( "uuid" ) for Url template ( "https://api.scryfall.com/cards/!set:uuid!?format=image&face=!prop:side!" ) is not available
PictureLoader: [card:  "Godhead"  set:  "CUBE" ]: Requested set property ( "muid" ) for Url template ( "https://api.scryfall.com/cards/multiverse/!set:muid!?format=image" ) is not available
PictureLoader: [card:  "Godhead"  set:  "CUBE" ]: Requested set property ( "muid" ) for Url template ( "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=!set:muid!&type=card" ) is not available
PictureLoader: [card:  "Godhead"  set:  "CUBE" ]: Trying to load picture
PictureLoader: [card:  "Godhead"  set:  "CUBE" ]: Picture not found on disk, trying to download
PictureLoader: [card:  "Godhead"  set:  "CUBE" ]: Trying to download picture from url: "MyCustomURL(removed)"
PictureLoader: [card:  "Godhead"  set:  "CUBE" ]: Image successfully downloaded from  "MyCustomURL(removed)"
libpng warning: iCCP: extra compressed data
libpng warning: iCCP: extra compressed data
